chain/network-primitives/src/types.rs

```diff
@@ -671,26 +669,27 @@ impl KnownPeerStatus {
 pub struct KnownPeerState {
     pub peer_info: PeerInfo,
     pub status: KnownPeerStatus,
-    pub first_seen: u64,
-    pub last_seen: u64,
+    pub first_seen: UnixTime,
+    pub last_seen: UnixTime,
 }
 
 impl KnownPeerState {
     pub fn new(peer_info: PeerInfo) -> Self {
+        let now = Time::now().to_unix_timestamp();
```

Comment:
:+1: for calling `now` once! 

Even better would be to pass `now` as an argument, but that's a separaet PR

Reply:
@matklad Yes, normally I would agree.

However, I checked all places which use `KnownPeerState::new`.
All those places call `new` exactly once, and the now wouldn't be useful for other purposes.

Reply:
>All those places call new exactly once,

Not really, here's one example where we call `new` in a loop, and already have time on our hand. 

```diff
diff --git a/chain/network/src/peer_manager/peer_store.rs b/chain/network/src/peer_manager/peer_store.rs
index 88e8a6dbf..209998715 100644
--- a/chain/network/src/peer_manager/peer_store.rs
+++ b/chain/network/src/peer_manager/peer_store.rs
@@ -58,6 +58,7 @@ impl PeerStore {
         let mut peer_states = HashMap::default();
         let mut addr_peers = HashMap::default();
 
+        let now = to_timestamp(Utc::now());
         for peer_info in boot_nodes.iter() {
             if !peer_states.contains_key(&peer_info.id) {
                 if let Some(peer_addr) = peer_info.addr {
@@ -70,7 +71,7 @@ impl PeerStore {
                             entry.insert(VerifiedPeer::signed(peer_info.id.clone()));
                             peer_states.insert(
                                 peer_info.id.clone(),
-                                KnownPeerState::new(peer_info.clone()),
+                                KnownPeerState::new(peer_info.clone(), now),
                             );
                         }
                     }
@@ -78,7 +79,6 @@ impl PeerStore {
             }
         }
 
-        let now = to_timestamp(Utc::now());
         for (key, value) in store.iter(ColPeers) {
             let key: Vec<u8> = key.into();
             let value: Vec<u8> = value.into();
```

Things like `add_indirect_peers` also seem to loop. 

Though, in any case this doesn't matter much.

Reply:
See https://github.com/near/nearcore/pull/5839/files